### DISCUSSION TITLE
[iOS] MASTG-TEST-0087 Apply Only to the Binary?

### DISCUSSION THREAD

**Discussion by barbieri-mobisec** [OWASP/mastg#2887] [Q&A] [answered]
Hello,
While following the MASTG, I came across this test `MASTG-TEST-0087: Make Sure That Free Security Features Are Activated`. In the static analysis part, it seems to focus primarily on the results of various FLAGS (PIE, Canary, ARC) related to the binary file. However, in the dynamic analysis part, **Objection** is used, but it’s not clear to me whether this should be employed exclusively to obtain the FLAG results for the main binary or if it should also be applied to all dylibs.

What do you think about this? Should dynamic analysis with Objection be extended to dynamic libraries as well, or should it remain focused on the main binary?

Maybe it would be helpful to make some changes to the test to clarify exactly what is expected.

**Comment by poffo-mobisec** [comment]
For what I understood, the test shall be applied as follow, identically between static and dynamic approach.

| **Analysis type** | **PIE**         | **Canary**              | **ARC**                 |
|-------------------|-----------------|-------------------------|-------------------------|
| Static            | `MH_EXECUTE` only | `MH_EXECUTE` and `MH_DYLIB` (not necessary on pure Swift apps) | `MH_EXECUTE` and `MH_DYLIB` (not necessary on pure C/C++ apps) |
| Dynamic           | `execute` only | `execute` and `dylib` (not necessary on pure Swift apps) | `execute` and `dylib` (not necessary on pure C/C++ apps) |

Please note: `execute` and `dylib` naming has been changed based on tool output to avoid any possible misunderstanding during test execution.

@cpholguera if my understanding of the test is correct, I think @barbieri-mobisec may suggest a pull request to clarify.

**Comment by poffo-mobisec** [comment, marked as answer]
Discussion can be closed due to [clarification on new MASTG-TEST-0228](https://github.com/OWASP/owasp-mastg/commit/e4e67c59c6318f96b4d72e24d32eeef11219e346).
